cmd/internal/codegen/parse/crd.go

```diff
@@ -410,9 +444,38 @@ func getValidation(comment string, props *v1beta1.JSONSchemaProps) {
 		}
 		props.MultipleOf = &f
 	case "Enum":
-		enums := strings.Split(parts[1], ",")
-		for i := range enums {
-			props.Enum = append(props.Enum, v1beta1.JSON{[]byte(enums[i])})
+		if props.Type != "array" {
+			value := strings.Split(parts[1], ",")
+			enums := []v1beta1.JSON{}
+			for _, s := range value {
+				// check type of enum element value to match type of field
+				// TODO support more types check
+				switch props.Type {
+				case "int", "int64", "uint64":
+					if _, err := strconv.ParseInt(s, 0, 64); err != nil {
+						log.Fatalf("Invalid integer value [%v] for a field of integer type", s)
+					}
+					enums = append(enums, v1beta1.JSON{[]byte(fmt.Sprintf("%v", s))})
+				case "int32", "unit32":
+					if _, err := strconv.ParseInt(s, 0, 32); err != nil {
+						log.Fatalf("Invalid integer value [%v] for a field of integer32 type", s)
+					}
+					enums = append(enums, v1beta1.JSON{[]byte(fmt.Sprintf("%v", s))})
+				case "float", "float32":
+					if _, err := strconv.ParseFloat(s, 32); err != nil {
+						log.Fatalf("Invalid float value [%v] for a field of float32 type", s)
+					}
+					enums = append(enums, v1beta1.JSON{[]byte(fmt.Sprintf("%v", s))})
+				case "float64":
+					if _, err := strconv.ParseFloat(s, 64); err != nil {
+						log.Fatalf("Invalid float value [%v] for a field of float type", s)
+					}
+					enums = append(enums, v1beta1.JSON{[]byte(fmt.Sprintf("%v", s))})
+				case "string":
+					enums = append(enums, v1beta1.JSON{[]byte(`"` + s + `"`)})
+				}
+			}
+			props.Enum = enums
 		}
 	case "Format":
 		props.Format = parts[1]
```

Comment:
All of these are perfect candidates for unit-test :) but that will require some refactoring of the code which is outside scope of this PR.

Reply:
Yes. I have encapsulated the handling logic into an individual util functions for better iteration in the future.